Dev_environment_instructions.md

```diff
@@ -0,0 +1,199 @@
+```
+git checkout tags/{{latest tag}}
+./gradlew publishToMavenLocal
+```
+When switching branch, you will see a message in console telling you which java version to use (you can switch it for your current shell with eg jenv). Then the needed jar should be in your m2 folder of your home directory. This will allow you to build certain images that require those .jar files.
+
+### yti-terminology-ui
+cd terminology-ui && ./build.sh
+
+### yti-fuseksi 
+Comment out the code *after* FROM statement and *until* the first COPY statement. Change FROM statement to  ```FROM stain/jena-fuseki:3.14.0```
+
+### yti-datamodel-api 
+For that you first had to build a correct image through yti-docker-java. 
+When in yti-docker-java directory, make sure you’re on the master branch and run ```docker build --platform linux/arm64 -t yti-docker-java17-base:corretto .``` to ensure that the image will be built for the M1 processor.
+
+After that, run ```./build.sh in yti-datamodel-api```
+
+### yti-datamodel-ui
+After you have built the correct java17 image for the M1 processor with ```docker build --platform linux/arm64 -t yti-datamodel-api .``` use ```./build.sh```
+
+*NB!!! (comment)* Should it be yti-datamodel-ui at the end of the docker build command? It's from my notes and I have that shell long closed, so I'm not sure about it
+
+### yti-comments-api
+```docker build -f docker/Dockerfile -t yti-comments-api  .```
+
+### yti-messaging-api
+```docker build -f docker/Dockerfile -t yti-messaging-api .```
+
+### yti-groupmanagement (?? instruction is too short for the amount of effort we poured in this image. But I can't recall anything else that would've been relevant)
+You would need node 14.18.2 installed locally. Change node-sass’ package version in package.json to 5.0.0 and set ```download = false``` in the ‘node’ object in the _build.gradle_ file. Then you rebuild the package
+
+### yti-codelist-ui
+Run the following command 
+``` yti-codelist-ui % docker build -f Dockerfile -t yti-codelist-ui . ```
+
+### yti-terminology-api
+You will need to have built spring jar 0.2.0 before running ./build.sh in yti-spring-security in the corresponding branch. Refer to the instruction above.
+
+
+### yti-postgres
+
+### yti-codelist-common-model 
+Contains jar files, a shared library
+
+### yti-codelist-public-api-service
+
+### yti-codelist-content-intake-service
+
+### yti-activemq 
+### yti-terminology-termed-docker
+
+*** at some point comment out checksum check in Dockerfile ***
+
+``` # && echo "${SHA} /tmp/apache-maven.tar.gz" | sha512sum -c - ```
+
+then run ```./build.sh```
+
+### yti-comments-api
+
+### yti-comments-ui
+
+
+## 3. After the images have been built – Docker final touches (is it needed?)
+
+### Changing images' names
+In docker-compose.yml, change names of images from eg *yti-example-name* to *docker-registry.rahti.csc.fi/mscr-test/yti-example-name:latest-arm64-v8*
```

Comment:
These instructions should work for all the platforms. Add a remark saying that the specific tags are needed for if running the local env with certain processor platforms.